docs/architecture/joins.txt

```diff
@@ -163,17 +159,123 @@ earlier:
 Consequently, the number of rows is significantly reduced allowing the
 operation to complete much faster.
 
+.. _joins_hash_join:
+
+Hash Join
+---------
+
+The Hash Join algorithm is used to execute certains types of joins in a more
+perfomant way than :ref:`Nested Loop <joins_nested_loop>`.
+
+Basic Algorithm
+...............
+
+The operation takes place in one node (the handler node to which the client is
+connected). The rows of the left relation of the join are read and a hashing
+algorithm is applied on the fields of that relation that participate in the
```

Comment:
maybe "on the fields of the relation which participate in the join" is a bit more clear (to avoid that that that)